src/OrbitQt/TrackTypeItemModel.cpp

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "TrackTypeItemModel.h"
+
+namespace orbit_qt {
+
+TrackTypeItemModel::TrackTypeItemModel(QObject* parent)
+    : QAbstractTableModel(parent), known_track_types_{Track::Type::kSchedulerTrack,
+                                                      Track::Type::kGpuTrack,
+                                                      Track::Type::kFrameTrack,
+                                                      Track::Type::kMemoryTrack,
+                                                      Track::Type::kThreadTrack,
+                                                      Track::Type::kVariableTrack} {}
+
+int TrackTypeItemModel::columnCount(const QModelIndex& parent) const {
+  return parent.isValid() ? 0 : static_cast<int>(Column::kEnd);
+}
+
+QVariant TrackTypeItemModel::data(const QModelIndex& idx, int role) const {
+  CHECK(idx.isValid());
+  CHECK(idx.model() == static_cast<const QAbstractItemModel*>(this));
+  CHECK(idx.row() >= 0 && idx.row() < static_cast<int>(known_track_types_.size()));
+  CHECK(idx.column() >= 0 && idx.column() < static_cast<int>(Column::kEnd));
+
+  Track::Type track_type = known_track_types_.at(idx.row());
+  Column col = static_cast<Column>(idx.column());
+
+  switch (role) {
+    case Qt::UserRole:
```

Comment:
`UserRole` is just an indicator of the first value you can use for your own roles, why is it being handled directly? If you need if for something you should define a constant and assign `Qt::UserRole` to it, so that you can give it a name, otherwise it's not clear what the actual role is.

Reply:
Changed